theano/tensor/elemwise.py

```diff
@@ -870,6 +870,13 @@ def perform(self, node, inputs, output_storage):
             elif (not isinstance(variable, numpy.ndarray) or
                   variable.dtype != nout.dtype):
                 variable = numpy.asarray(variable, nout.dtype)
+                # The next line is needed for numpy 1.9. Otherwise
+                # there are tests that fail in DebugMode.
+                # Normally we would call theano.misc._asarray, but it
+                # is faster to inline the code. We know that the dtype
+                # are the same string, just different typenum.
+                if numpy.dtype(nout.dtype).num != variable.dtype.num:
+                    variable = variable.view(dtype=nout.dtype)
```

Comment:
Ok then, but how are we certain that this won't do bad views (like int64 to uint64)?

Or put another way, how do we know that the dtype is the same string?